pcstac/tests/conftest.py

```diff
@@ -43,75 +17,99 @@ def event_loop():
-@pytest.fixture(scope="session")
-async def app_client(app):
-    async with AsyncClient(app=app, base_url="http://test") as c:
-        yield c
-
-
-@pytest.fixture
-def load_test_data() -> Callable[[str], Dict]:
-    def load_file(filename: str) -> Dict:
-        with open(os.path.join(DATA_DIR, filename)) as file:
-            return json.load(file)
-
-    return load_file
+async def app_client():
+    """Setup DB and application."""
+    print("Setting up test db")
+    with pytest.MonkeyPatch.context() as mp:
+        mp.setenv("POSTGRES_DBNAME", "pgstactestdb")
+
+        # Setting this environment variable to ensure links are properly constructed
+        mp.setenv("TILER_HREF_ENV_VAR", "http://localhost:8080/stac/dqe")
+
+        from stac_fastapi.pgstac.config import Settings
+        from stac_fastapi.pgstac.db import close_db_connection, connect_to_db
+
+        from pcstac.main import app
+
+        # Testing is set to false because creating/updating via the API is not desirable
+        #  thus, we actually want to use the migrations and data loaded in via the setup
+        #  script which builds PQE docker containers
+        settings = Settings(testing=False)
+        assert settings.postgres_dbname == "pgstactestdb"
+
+        # First connect to the default DB (which should be `postgres`)
+        defaut_dsn = settings.reader_connection_string.replace(
+            "/pgstactestdb", "/postgres"
+        )
+        print(f"Connecting to write database {defaut_dsn}")
+        print("writer conn string", defaut_dsn)
+        conn = await asyncpg.connect(dsn=defaut_dsn)
+
+        print("creating temporary `pgstactestdb` database...")
+        try:
+            await conn.execute("CREATE DATABASE pgstactestdb;")
+            await conn.execute(
+                "ALTER DATABASE pgstactestdb SET search_path to pgstac, public;"
+            )
+        except asyncpg.exceptions.DuplicateDatabaseError:
+            print("pgstactestdb already exists, cleaning it...")
+            await conn.execute("DROP DATABASE pgstactestdb;")
+            await conn.execute("CREATE DATABASE pgstactestdb;")
+            await conn.execute(
+                "ALTER DATABASE pgstactestdb SET search_path to pgstac, public;"
+            )
+        finally:
+            await conn.close()
+
+        try:
+            print("migrating...")
+            conn = await asyncpg.connect(dsn=settings.reader_connection_string)
+            val = await conn.fetchval("SELECT true")
+            assert val
+            await conn.close()
+
+            version = await pypgstac.run_migration(
+                dsn=settings.reader_connection_string
+            )
+            print(f"PGStac Migrated to {version}")
+
+            print("Registering collection and items")
+            conn = await asyncpg.connect(dsn=settings.reader_connection_string)
+            await conn.copy_to_table(
+                "collections",
+                source=collections,
```

Comment:
I had to update the collections and items files to make it work with `copy_to_table`